File: Project 1/blink.c
//Name: Khan, Muhammad
//Date: 3/13/2021

#include <wiringPi.h>
#include <signal.h>

#define RED 23
#define REDT 24
#define REDTH 25

int blink = 0;

void cleanup(int signo) {
     blink = 0;
}

int main(void) {
     signal(SIGINT, cleanup);
     signal(SIGTERM, cleanup);
     signal(SIGHUP, cleanup);

    wiringPiSetup();
    pinMode(RED, OUTPUT);
    pinMode(REDT, OUTPUT);
    pinMode(REDTH, OUTPUT);

    while(blink) {
        digitalWrite(RED, HIGH);
        delay(500);
        digitalWrite(RED, LOW);
        digitalWrite(REDT, HIGH);
        delay(500);
        digitalWrite(REDT, LOW);
        digitalWrite(REDTH, HIGH);
        delay(500);
        digitalWrite(REDTH, LOW);
      }

      digitalWrite(RED, LOW);
      digitalWrite(REDT, LOW);
      digitalWrite(REDTH, LOW);

      pinMode(RED, INPUT);
      pinMode(REDT, INPUT);
      pinMode(REDTH, INPUT);
      return 0;
}

File: Project 1/blink.py
#!/usr/bin/python3

"""
Name: Khan, Muhammad
Date: 3/13/2021
"""
from gpiozero import LED
from signal import pause 
from time import sleep

led1 = LED(13)
led2 = LED(19)
led3 = LED(26)

try:
   led1.blink(1,3)
   sleep(1)
   led2.blink()
   sleep(1)
   led3.blink(1,3)

   pause()

except KeyboardInterrupt:
    pass
